Index: src/org/eclipse/dltk/ui/text/AbstractScriptScanner.java
===================================================================
RCS file: /cvsroot/technology/org.eclipse.dltk/core/plugins/org.eclipse.dltk.ui/src/org/eclipse/dltk/ui/text/AbstractScriptScanner.java,v
retrieving revision 1.2
diff -u -r1.2 AbstractScriptScanner.java
--- src/org/eclipse/dltk/ui/text/AbstractScriptScanner.java	2 May 2007 15:07:29 -0000	1.2
+++ src/org/eclipse/dltk/ui/text/AbstractScriptScanner.java	9 Jun 2007 16:40:15 -0000
@@ -14,6 +14,7 @@
 import java.util.List;
 import java.util.Map;
 
+import org.eclipse.core.runtime.Assert;
 import org.eclipse.dltk.ui.PreferenceConstants;
 import org.eclipse.jface.preference.IPreferenceStore;
 import org.eclipse.jface.preference.PreferenceConverter;
@@ -351,4 +352,16 @@
 	protected IPreferenceStore getPreferenceStore() {
 		return fPreferenceStore;
 	}
+	
+	/**
+	 * @return never <code>null</code>.
+	 */
+	public TextAttribute getTextAttribute(String key) {
+	    Token token = getToken(key);
+	    Assert.isNotNull(token);
+	    TextAttribute attr = (TextAttribute) token.getData();
+	    Assert.isNotNull(attr);
+	    return attr;
+	}
+	
 }
